app/controllers/users_controller.rb

```diff
@@ -9,6 +9,7 @@ def destroy
 
   def destroy_user
     user = current_user || User.find_by(confirmation_token: session[:user_confirmation_token])
+    session.delete(:user_confirmation_token)
```

Comment:
Totally! I didn't mean to imply that it wasn't, I didn't consider the sp_session behavior at first. I only noticed it when I ran the specs. 

It seems like deleting the whole session is preferable because, as you said, each visitor gets them. In the event we add other one-off keys to the session, we don't want to forget to delete them and have them persist when another user tries to use the service (if the users were on a shared computer or something).